Keep LHE angles/probs in the Events tree only; Reduce/standardize verbosity; minor fixes

namapane opened the pull request:
- Move LHEAngProbFiller to the normal sequence, without adding it to addAllEvents. 
- Fix a bug in filling angles in cases where `hMass - daughters.MTotal()) < 0.01`:  since no `fillBranch `was called in that case, the values from the previous events were taken
- Some improvements to printouts and implementation.

Notes:
- Updated [exampleProbabilities.py](https://github.com/CJLST/ZZAnalysis/pull/352/files#diff-b64636862a030d2b24ed1f3cdb436f5c584f037bde4923c5bc8bb0325e378343R37-R39) for the documentation of "`addPAux`", "`addPmavjj`" , "`addPmavjj_true`". (Documentation should be moved elsewhere at some point)

namapane committed:
reduce verbosity; minor improvements

namapane commented:
(@pianonick411 you may want to have a look to the changes and to the question above in the meanwhile)

pianonick411 commented on NanoAnalysis/python/MELAProbHelper.py:
```diff
@@ -192,42 +189,41 @@ def fillProbs(self, candDaughters, candAssociated, candMothers):
-                    self.out.fillBranch(MELA_branchname, probVec)
+                        probPropVec = probPropVec[0]
+                    if MELA_addPAux:
+                        probVec_PAux - probVec_PAux[0]
```
Is L212 meant to be `probVec_PAux  = probVec_PAux[0]`?

namapane replied:
Right! thanks for spotting this.

pianonick411 commented:
I have found [where pAux is calculated](https://spin.pha.jhu.edu/MELA/Mela_8cc_source.html#l01715) in the c++ interface to MELA. 

`mela.getPAux` seems to only retrieve some internal variable `auxiliaryprob`.  See [here](https://spin.pha.jhu.edu/MELA/Mela_8cc_source.html#l00594). 

It seems that when computeProdP is called for processes that are not VH, the value of `auxiliaryprob` is modified. In the linked presentation, the probability denoted with `pAux` is defined with an integral over eta. I believe that the "integration" over eta is evaluated numerically [here](https://spin.pha.jhu.edu/MELA/Mela_8cc_source.html#l01729) and has the effect of integrating out one jet. `auxiliaryprob` ends up taking on the result of this computation.  

Hopefully this clears things up at least somewhat.

namapane committed:
fix bug affecting LHE angle for events not matching the H mass

namapane commented:
@pianonick411, in the [last commit](https://github.com/CJLST/ZZAnalysis/pull/352/commits/2c1eab44c5973974721a5695dc1d09aa780a90d4) I added a small fix since I observed an event which fails the condition `abs(hMass - daughters.MTotal()) < 0.01` (it is event 802250 in the MELA_Test file). The problem is that variables should be filled in any case, also in pathological conditions, otherwise the values from the previous event remain in the ttree buffer and are used for the current event as well.
The `elif` [here](https://github.com/CJLST/ZZAnalysis/pull/352/commits/2c1eab44c5973974721a5695dc1d09aa780a90d4#diff-ae5c74aabd9843c6bf0db36723b4e202f99590b7d1e01eb748500bc2cea1f085L168) had "<" instead of ">" so that also prevented the warning message to be printed.
This fix may become irrelevant if we move angles computation to `MELAProbHelper` but it's a good idea to fix this right away.

I also added [filling of `qH, mZ1, mZ2`](https://github.com/CJLST/ZZAnalysis/pull/352/commits/2c1eab44c5973974721a5695dc1d09aa780a90d4#diff-ae5c74aabd9843c6bf0db36723b4e202f99590b7d1e01eb748500bc2cea1f085R167-R169), which were booked but not filled; are they actually needed? If not, we could remove these variables altogether.

pianonick411 commented:
@namapane, I am not sure what is best to do with `qh, mZ1` and `mZ2`. They are useful for debugging MELA, but otherwise only duplicate information that is already present. I lean more towards getting rid of them. If we need to debug, dumping the MELA computation using an increased verbosity is much more comprehensive, anyways.

namapane committed:
Remove LHEAngProbFiller variables from AllEvents tree

namapane committed:
update doc

namapane committed:
restore

namapane commented:
> @namapane, I am not sure what is best to do with `qh, mZ1` and `mZ2`. They are useful for debugging MELA, but otherwise only duplicate information that is already present. I lean more towards getting rid of them. If we need to debug, dumping the MELA computation using an increased verbosity is much more comprehensive, anyways.

I eventually commented them out; they can be added back when needed.
I also moved LHEAngProbFiller out of the sequence filling the `AllEvents `tree, so that it is run only for selected events and results are stored only in the `Events ` tree. This significantly reduces processing time and disk space. 
We can of course revert this if a need to have some of the variables in the `AllEvents` tree comes up.
Small note: this change has a small side effect, that is, LHEAngProbFiller would now be called also for background samples (whereas it was called only for signals before, as a side effect of the fact that ADD_ALLEVENTS is generally set True only for signals). I will fix this in a separate PR.
